network/p2p/throttler_test.go

```diff
@@ -0,0 +1,155 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package p2p
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+func TestSlidingWindowThrottlerHandle(t *testing.T) {
+	period := time.Minute
+	previousWindowStartTime := time.Time{}
+	currentWindowStartTime := previousWindowStartTime.Add(period)
+
+	nodeID := ids.GenerateTestNodeID()
+
+	type call struct {
+		time      time.Time
+		throttled bool
+	}
+
+	tests := []struct {
+		name  string
+		limit int
+		calls []call
+	}{
+		{
+			name:  "throttled in current window",
+			limit: 1,
+			calls: []call{
+				{
+					time: currentWindowStartTime,
+				},
+				{
+					time:      currentWindowStartTime,
+					throttled: true,
+				},
+			},
+		},
+		{
+			name:  "throttled from previous window",
+			limit: 2,
+			calls: []call{
+				{
+					time: previousWindowStartTime,
+				},
+				{
+					time: previousWindowStartTime.Add(period),
+				},
+				{
+					time:      currentWindowStartTime.Add(time.Second),
+					throttled: true,
+				},
+			},
+		},
+		{
+			name:  "throttled over multiple evaluation periods",
+			limit: 5,
+			calls: []call{
+				{
+					time: currentWindowStartTime.Add(30 * time.Second),
```

Comment:
I originally did this but found it was just easier to do seconds explicitly since when debugging it was easier to tell which call was at which point in time.